src/main/kotlin/org/rust/ide/inspections/lints/RsUnusedLabelsInspection.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import com.intellij.psi.PsiElement
+import org.rust.RsBundle
+import org.rust.ide.annotator.fixes.RemoveElementFix
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.lang.core.psi.RsLabelDecl
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.searchReferences
+
+/** Analogue of rustc's unused_labels. */
+class RsUnusedLabelsInspection : RsLintInspection() {
+    override fun getLint(element: PsiElement) = RsLint.UnusedLabels
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean) = object : RsVisitor() {
+        override fun visitLabelDecl(o: RsLabelDecl) {
+            val isLabelUsed = o.searchReferences().any()
```

Comment:
```suggestion
            val isLabelUsed = o.searchReferences().findFirst() != null
```
Note, `com.intellij.util.Query` is a non-lazy iterator, so calling `.any()` leads to collecting all the references instead of picking just one